### PR TITLE
Super flexible Tile components

### PR DESCRIPTION
**Pull request by jstnjns**
## Overview
There's a fine balance between building a unique tile for every need/context, and having generalized tiles that have wayyyy too many props to get them to look like they need to in unique cases.

I submit to you, the start of a collection of composable tile elements that allow you to choose from a collection of common parts that make up tiles, but don't limit you to what is already built into the component.  For instance, if you are building a tile and the designer has introduced a new block of text that current tiles don't support you'd have two options: build the tile from scratch (even though maybe 80% of the pattern exists in existing components) or hack away at an existing component with conditional props to get your bespoke solution.

screen shot 2018-07-25 at 11 50 43 pm
screen shot 2018-07-25 at 11 50 47 pm
screen shot 2018-07-25 at 11 50 33 pm
screen shot 2018-07-25 at 11 50 38 pm

## Risks
ImageTile is currently used.  However, I didn't touch ImageTile, but instead created one that sounds similar (TileImage).  We should eventually phase out ImageTile because a combination of Tile/TileImage/TileOverlay/TileCaption does the exact same thing but with more flexibility.

## Changes
No changes to existing components, but a change to the way we build tiles in the future.

## Issue
This PR makes #66 irrelevant because we'd just deprecate ImageTile altogether.

### CODE REVIEW

File: src/components/Tile/index.stories.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import TileCaption from '../TileCaption'
+import TileCheck from '../TileCheck'
+import AnimationHandler from '../AnimationHandler'
+import HoverHandler from '../HoverHandler'
+
+import shondaRhimesThumbnail from '../../assets/shonda-rhimes-video.png'
+
+
+storiesOf('components|Tiles', module)
+  .add('Summary', () => (
```
Comment (jstnjns) [inline src/components/Tile/index.stories.js]: There's a lot in this PR, but this story should hopefully explain most of it.

**Comment by jstnjns** [comment]
It wouldn't let me post this gif last night.  But here's an example of how you can stack the parts to make a whole.

tile

**Comment by jstnjns** [comment]
Here's some direction I already see:

TileVideo gets added
TileOverlay supports more types (fill, gradient-top, etc)
TileCaption takes typography options (title-level="h2") and positioning props (left-bottom, center-bottom, center)

**Comment by stephenp** [comment]
I don't feel qualified to weigh in on a change this large, but the class name implementations for hover and animation events looks good to me.

**Pull request merged**